packages/next/src/build/index.ts

```diff
@@ -948,6 +949,12 @@ export default async function build(
           )
         )
       loadedConfig = config
+      if (config.experimental.lockDistDir) {
+        await Lockfile.acquireOrExit(
```

Comment:
I mean that works, but it might not be super clear.

* Rust acquires the lockfile and returns an External object pointing to the `Mutex<ManuallyDrop<Option<File>>>`
* We don't store the Lockfile anyway, so it's GCed
* This drops the `Mutex<ManuallyDrop<Option<File>>>` object, which forgets about the `Option<File>`
* So the file handle stays open, but there is no reference to it anywere.
* When the process exits, OS closes all file handles, because it always does so.
* When the file handle is closed, it unlocks the file.

If that behavior is intentional, add some documentation to somewhere. Maybe to `JsLockfile`.

Reply:
I left the leaking behavior on `build`, but added some cleanup on `dev` in case somebody is using a custom server that creates multiple instances of the server, in which case we should release the lock on `close`.